appeals/web/src/server/appeals/appeal-details/appellant-case/appellant-case.controller.js

```diff
@@ -0,0 +1,272 @@
+import logger from '../../../lib/logger.js';
+import * as appealDetailsService from '../appeal-details.service.js';
+import * as appellantCaseService from './appellant-case.service.js';
+import {
+	mapResponseToSummaryListBuilderParameters,
+	mapWebReviewOutcomeToApiReviewOutcome,
+	mapReviewOutcomeToSummaryListBuilderParameters,
+	mapReviewOutcomeToNotificationBannerComponentParameters
+} from './appellant-case.mapper.js';
+import { generateSummaryList } from '../../../lib/nunjucks-template-builders/summary-list-builder.js';
+import { objectContainsAllKeys } from '../../../lib/object-utilities.js';
+import { renderDecisionValidConfirmationPage } from './outcome-valid/outcome-valid.controller.js';
+import { renderDecisionInvalidConfirmationPage } from './outcome-invalid/outcome-invalid.controller.js';
+import { renderDecisionIncompleteConfirmationPage } from './outcome-incomplete/outcome-incomplete.controller.js';
+import { appellantCaseReviewOutcomes } from '../../appeal.constants.js';
+
+/**
+ *
+ * @param {import('@pins/express/types/express.js').Request} request
+ * @param {import('@pins/express/types/express.js').RenderedResponse<any, any, Number>} response
+ */
+const renderAppellantCase = async (request, response) => {
+	const { errors } = request;
+
+	const appealDetails = await appealDetailsService
+		.getAppealDetailsFromId(request.params.appealId)
+		.catch((error) => logger.error(error));
+
+	if (appealDetails) {
+		const appealReferenceFragments = appealDetails?.appealReference.split('/');
+		const formattedSiteAddress = appealDetails?.appealSite
+			? Object.values(appealDetails?.appealSite)?.join(', ')
+			: 'Address not known';
+
+		const appellantCaseResponse = await appellantCaseService
+			.getAppellantCaseFromAppealId(appealDetails?.appealId, appealDetails?.appellantCaseId)
+			.catch((error) => logger.error(error));
+
+		const mappedAppellantCaseSections =
+			mapResponseToSummaryListBuilderParameters(appellantCaseResponse);
+		const formattedSections = [];
+		for (const section of mappedAppellantCaseSections) {
+			formattedSections.push(generateSummaryList(section.rows, section.header));
+		}
+
+		let notificationBannerParameters;
+
+		if (objectContainsAllKeys(request.session, 'appellantCaseReviewOutcomeNotificationData')) {
+			// @ts-ignore
+			const notificationData = request.session.appellantCaseReviewOutcomeNotificationData;
+			const reasonOptions = await appellantCaseService.getAppellantCaseNotValidReasonsForOutcome(
+				notificationData.validationOutcome
+			);
+			if (!reasonOptions) {
+				throw new Error('error retrieving invalid reason options');
+			}
+
+			notificationBannerParameters = mapReviewOutcomeToNotificationBannerComponentParameters(
+				reasonOptions,
+				notificationData.validationOutcome,
+				notificationData.invalidOrIncompleteReasons,
+				notificationData.otherNotValidReasons
+			);
+
+			// @ts-ignore
+			delete request.session.appellantCaseReviewOutcomeNotificationData;
```

Comment:
It is intentional, but on checking the ticket, there's no mention that the notification should disappear after the first view. That was an assumption on my part. I've removed the code related to displaying this banner now, as I realised it wasn't working as intended. As you mentioned, we really need the review outcome, reasons etc. to be returned from the API with the rest of the appellant case data, because using the session as I was originally isn't going to work. I've left the supporting code used to generate and render the banner, as that will be needed once the API has been updated.